pybamm/solvers/scikits_dae_solver.py

```diff
@@ -43,7 +43,7 @@ def method(self):
     def method(self, value):
         self._method = value
 
-    def integrate(self, residuals, y0, t_eval, events=None):
+    def integrate(self, residuals, y0, t_eval, events=None, mass_matrix=None):
```

Comment:
For finite volumes the mass matrix is just the identity, but for other schemes this may be different. Jamie has some specific implementations of finite elements and control volumes where the mass matrix is different, and this may be a way around needing a separate strong and weak form for each equation. From what I can tell you can't pass a mass matrix to scipy odeint -- if you want to use that for problems of the form M*y_dot = f(t,y) you have to invert M first, which I guess is not ideal. You can pass the mass matrix into scikits.odes if using IDA, but it is wrapped up with jacobian so can't be put in until that is working. In this case, maybe it is better to not pass this to all solvers?

Reply:
no, probably leave it as is. even if the solvers we currently have don't support it, they all could (in theory) use the mass matrix for time integration, so it should be part of the interface. I'm happy to merge it as is